torch/csrc/jit/codegen/cuda/python_frontend/fusion_record.h

```diff
@@ -256,6 +258,12 @@ struct InputTensorRecord : RecordFunctor {
                   .dtype(dtype)
                   .build();
 
+    if (symbolic_sizes.empty() && is_cpu) {
+      tv->setCpuScalar(true);
+    } else {
+      TORCH_INTERNAL_ASSERT(!is_cpu, "cpu non-scalar tensor is not supported");
```

Comment:
This should be `TORCH_CHECK` instead of `TORCH_INTERNAL_ASSERT` because the assert is something that's unexpected and the error message suggest to the user to file a bug report.

When we have the following graph:
```
# gm.graph.print_tabular()
opcode         name    target               args        kwargs
-------------  ------  -------------------  ----------  --------
placeholder    x_1     x_1                  ()          {}
placeholder    y_1     y_1                  ()          {}
call_function  add     nvprims.add.default  (x_1, y_1)  {}
output         output  output               (add,)      {}
```
We'd get the following error message:
```py
# nvfuser_execute(gm, a, a.cpu())
RuntimeError: !is_cpu INTERNAL ASSERT FAILED at "/home/iyashchuk/dev/pytorch/master/torch/csrc/jit/codegen/cuda/python_frontend/fusion_record.h":264, please report a bug to PyTorch. cpu non-scalar tensor is not supported
```